shotover-proxy/src/transforms/redis/sink_cluster.rs

```diff
@@ -124,47 +122,56 @@ impl RedisSinkCluster {
         };
 
         let routing_info = RoutingInfo::for_command_frame(command)?;
-        let channels = match self.get_channels(routing_info) {
-            ChannelResult::Channels(channels) => channels,
-            ChannelResult::Auth => {
-                return self.on_auth(command).await;
-            }
-            ChannelResult::ShortCircuit(frame) => {
-                let (one_tx, one_rx) = immediate_responder();
-                send_frame_response(one_tx, frame)
-                    .map_err(|_| anyhow!("Failed to send short circuited redis frame"))?;
-                return Ok(Box::pin(one_rx));
-            }
-        };
+        self.dispatch_message_hiding(routing_info, message).await
+    }
 
-        Ok(match channels.len() {
+    async fn send_message_to_slot(
+        &mut self,
+        slot: u16,
+        message: Message,
+    ) -> Result<ResponseFuture> {
+        if let Some((_, lookup)) = self.slots.masters.range(&slot..).next() {
+            let lookup = lookup.to_string();
+            let one_rx = self.choose_and_send(&lookup, message).await?;
+            Ok(Box::pin(
+                one_rx.map_err(|_| anyhow!("no response from single channel")),
+            ))
+        } else {
+            self.send_error_response(
+                self.reason_for_no_nodes
+                    .unwrap_or("ERR Shotover RedisSinkCluster does not know of a node containing the required slot")
+            )
+        }
+    }
+
+    async fn send_message_to_channels(
+        &mut self,
+        channels: &[String],
+        message: Message,
+        routing_info: RoutingInfo,
+    ) -> Result<ResponseFuture> {
+        match channels.len() {
             // Return an error as we cant send anything if there are no channels.
-            0 => {
-                let (one_tx, one_rx) = immediate_responder();
-                match self.reason_for_no_nodes {
-                    Some(message) => {
-                        self.send_error_response(one_tx, message).ok();
-                    }
-                    None => self.short_circuit(one_tx),
-                };
-                Box::pin(one_rx)
-            }
+            0 => self.send_error_response(
+                self.reason_for_no_nodes
+                    .unwrap_or("ERR Shotover RedisSinkCluster does not have any nodes"),
```

Comment:
I wouldn't expect to ever see this, but "cluster does not have any nodes" would be confusing (impossible?). 😂

Reply:
Well it is `RedisSinkCluster does not have any nodes` which should have a different meaning.
Does it make more sense like that?

Or do you have a recommendation for a better error? (The current one isnt great but not sure how to improve it)

Reply:
Maybe `RedisSinkCluster does not know of any nodes` makes it more clear?

Reply:
I went ahead and changed it to `RedisSinkCluster does not know of any nodes`.
I'll queue the merge as is.